File: platform/processing-core/app/schemas/partner_onboarding.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class PartnerOnboardingPartnerOut(BaseModel):
    id: str
    code: str
    legal_name: str
    brand_name: str | None = None
    partner_type: str
    status: str
    contacts: dict[str, Any] = Field(default_factory=dict)


class PartnerOnboardingChecklistOut(BaseModel):
    profile_complete: bool
    legal_documents_accepted: bool
    legal_profile_present: bool
    legal_details_present: bool
    legal_details_complete: bool
    legal_verified: bool
    activation_ready: bool
    blocked_reasons: list[str] = Field(default_factory=list)
    next_step: str
    next_route: str


class PartnerOnboardingSnapshotOut(BaseModel):
    partner: PartnerOnboardingPartnerOut
    checklist: PartnerOnboardingChecklistOut


class PartnerOnboardingProfilePatch(BaseModel):
    brand_name: str | None = None
    contacts: dict[str, Any] | None = None


__all__ = [
    "PartnerOnboardingChecklistOut",
    "PartnerOnboardingPartnerOut",
    "PartnerOnboardingProfilePatch",
    "PartnerOnboardingSnapshotOut",
]

File: frontends/partner-portal/src/api/partnerOnboarding.ts
import { request } from "./http";

export interface PartnerOnboardingSnapshot {
  partner: {
    id: string;
    code: string;
    legal_name: string;
    brand_name?: string | null;
    partner_type: string;
    status: string;
    contacts?: Record<string, unknown>;
  };
  checklist: {
    profile_complete: boolean;
    legal_documents_accepted: boolean;
    legal_profile_present: boolean;
    legal_details_present: boolean;
    legal_details_complete: boolean;
    legal_verified: boolean;
    activation_ready: boolean;
    blocked_reasons: string[];
    next_step: string;
    next_route: string;
  };
}

export const fetchPartnerOnboarding = (token: string) =>
  request<PartnerOnboardingSnapshot>("/partner/onboarding", { method: "GET" }, { token, base: "core_root" });

export const patchPartnerOnboardingProfile = (
  token: string,
  payload: { brand_name?: string; contacts?: Record<string, unknown> },
) =>
  request<PartnerOnboardingSnapshot["partner"]>(
    "/partner/onboarding/profile",
    { method: "PATCH", body: JSON.stringify(payload) },
    { token, base: "core_root" },
  );

export const activatePartnerOnboarding = (token: string) =>
  request<PartnerOnboardingSnapshot>("/partner/onboarding/activate", { method: "POST" }, { token, base: "core_root" });
